compiler/packages/babel-plugin-react-compiler/src/Optimization/ConstantPropagation.ts

```diff
@@ -509,6 +509,72 @@ function evaluateInstruction(
       }
       return null;
     }
+    case 'TemplateLiteral': {
+      if (value.subexprs.length === 0) {
+        const result: InstructionValue = {
+          kind: 'Primitive',
+          value: value.quasis.map(q => q.cooked).join(''),
+          loc: value.loc,
+        };
+        instr.value = result;
+        return result;
+      }
+
+      if (value.subexprs.length !== value.quasis.length - 1) {
+        return null;
+      }
+
+      if (value.quasis.some(q => q.cooked === undefined)) {
+        return null;
+      }
+
+      let quasiIndex = 0;
+      let resultString = value.quasis[quasiIndex].cooked as string;
+      ++quasiIndex;
+
+      for (const subExpr of value.subexprs) {
+        const subExprValue = read(constants, subExpr);
+        if (!subExprValue || subExprValue.kind !== 'Primitive') {
+          return null;
+        }
+
+        const expressionValue = subExprValue.value;
+        if (
+          (typeof expressionValue === 'object' && expressionValue !== null) ||
+          typeof expressionValue === 'symbol' ||
+          typeof expressionValue === 'function'
+        ) {
```

Comment:
hmm, let's allowlist instead of blocklist.

Reply:
Very good point! Done in cee8ece0ff753b38dea34f40ccdd25597a9bb1c8